s3buildcache/src/main/kotlin/androidx/build/gradle/s3buildcache/S3StorageService.kt

```diff
@@ -101,11 +107,13 @@ class S3StorageService(
 
     override fun validateConfiguration() {
         try {
-            val buckets = client.listBuckets().buckets()
-            if (buckets.none { bucket -> bucket.name() == bucketName }) {
-                throw Exception("Bucket $bucketName under project $region cannot be found or is not accessible using the provided credentials")
-            }
-        } catch (e: Exception) {
+            client.headBucket(HeadBucketRequest.builder().bucket(bucketName).build())
```

Comment:
This is a good change, thanks. The reason I chose to fail silently is I didn't know these exception subtypes existed, and what was happening is projects just wouldn't build when offline because the AWS SDK would always fail the validation step. I'm suspicious of `SdkException` down below. Can you confirm this exception isn't thrown if the host machine isn't connected to the internet? Thanks

Reply:
I see, sure I can check offline operations.

For offline operations though, wouldn't throwing BuildCacheException offer a better experience (gradle disables the cache then and prints a "due to errors, cache was disabled" at the end of the build)?

Reply:
>For offline operations though, wouldn't throwing BuildCacheException offer a better experience (gradle disables the cache then and prints a "due to errors, cache was disabled" at the end of the build)?

you'd think, but IIRC Gradle just gives up and stops building

Reply:
> you'd think, but IIRC Gradle just gives up and stops building

interesting, I've seen the cache being disabled gracefully when I run it against a server that returned http 403s due to misconfiguration. I guess I can check that as well.

Reply:
in any case, you were correct, the plugin would thrown when offline. I've narrowed down the exception scope to SdkServiceExceptions, connectivity related issues should inherit from https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/exception/SdkClientException.html and those 2 are siblings. 

Also tested manually and the build only prints a warning when I disconnect from wifi.

I figure using BuildCacheExceptions as opposed to validate method should be a separate PR anyway so I'll hold on on doing that if it's ok with you